src/processout/processout.ts

```diff
@@ -638,5 +690,77 @@ module ProcessOut {
             var handler = new ActionHandler(this, options);
             return handler.handle(url, success, error);
         }
+
+        /**
+         * MakeCardPayment makes a full card payment, handling any required
+         * customer action such as authentication for SCA (like 3DS 1 or 2)
+         * @param {string} invoiceID
+         * @param {string} cardID 
+         * @param {any} options 
+         * @param {callback} success 
+         * @param {callback} error
+         */
+        public makeCardPayment(invoiceID: string, cardID: string,
+            options: any,
+            success:  (data: any)       => void, 
+            error:    (err:  Exception) => void): void {
+
+            if (!options) options = {};
+
+            var source = cardID;
+            if (options.gatewayRequestSource)
+                source = options.gatewayRequestSource;
+
+            this.apiRequest("POST", `invoices/${invoiceID}/capture`, {
+                "authorize_only": options.authorize_only,
+                "capture_amount": options.capture_amount,
+                "source":         source,
+
+                "enable_three_d_s_2": true
+            }, function(data: any): void {
+                if (!data.success) {
+                    error(new Exception(data.error_type, data.message));
+                    return;
+                }
+
+                if (!data.customer_action) {
+                    success(invoiceID);
+                    return;
+                }
+
+                var nextStep = function(data: any): void {
+                    options.gatewayRequestSource = data;
+                    this.makeCardPayment(invoiceID, cardID, options, success, error);
+                }.bind(this);
+
+                switch (data.customer_action.type) {
+                case "url":
```

Comment:
As discussed we will try to deprecate `url` and eventually merge it with `redirect`, right?

Reply:
Yep